pkg/k8s/apis/cilium.io/client/crds/v2/ciliumclusterwidenetworkpolicies.yaml

```diff
@@ -1563,6 +1563,32 @@ spec:
                             type: array
                         type: object
                       type: array
+                    icmp6:
```

Comment:
Agreed, I think this is a good idea.

Reply:
@pchaigno use `family` field like this right?
```yaml
...
spec:
  ingress:
  - fromEndpoints:
    - matchLabels:
        org: empire
    toICMPs:
      family:
        toICMPs:
        - types:
          - type: 8
        toICMP6s:
        - types:
          - type: 128
```

Reply:
I was thinking of something like:
```
spec:
  ingress:
  - fromEndpoints:
    - matchLabels:
        org: empire
    toICMPs:
    - icmps:
      - family: IPv4
        type: 8
      - family: IPv6
        type: 128
```
Not sure what should be the value for `family:`. Maybe `ICMP` vs. `ICMPv6` is more appropriate?

Also, we could make `family: IPv4` the default for brevity.

Reply:
I think the unwrapped format is better.
As you said, we can reduce codes by wrapping them, but I like current separated style because it is simpler and easy to read.

Reply:
Then maybe we can find some other way to reduce code duplication?

Reply:
Ah... to reduce codes I think your idea is the best.
In my thought, ICMP and ICMPv6 are completely different things, so I wanted to separate. But I'm starting to think wrapped style is good.
Introduce `family` field makes the manifest complicated but not so much and the amount of codes we'll reduce is so much.

Reply:
And if it defaults to IPv4 family, then it shouldn't be more complicated for the most common cases. I agree ideally it might be best to separate ICMP and ICMPv6 because they are different protocols but at the same time they both have types and codes :shrug: One could argue it's the same as having UDP and TCP under the same `toPorts` field.

If in doubt, we can always raise it on e.g. #sig-policy or #networkpolicy on Slack.